airflow/kubernetes/pod_generator.py

```diff
@@ -430,6 +419,70 @@ def construct_pod(
         except Exception as e:
             raise PodReconciliationError from e
 
+    @classmethod
+    def build_selector_for_k8s_executor_pod(
+        cls,
+        *,
+        dag_id,
+        task_id,
+        try_number,
+        map_index=None,
+        date=None,
+        run_id=None,
+    ):
+        """
+        Generate selector for kubernetes executor pod
+
+        :meta private:
+        """
+        labels = cls.build_labels_for_k8s_executor_pod(
+            dag_id=dag_id,
+            task_id=task_id,
+            try_number=try_number,
+            map_index=map_index,
+            date=date,
```

Comment:
Do we need both execution_date and run_id on the selector?

Reply:
well their both optional -- you could use either or both or none...

Reply:
That surprises me a bit, run_id is in the TI pk. That might have wider impact though...

Reply:
so, it's probably backcompat legacy for construct_pod... because everything is positional args they are all "required"